### ISSUE TITLE
Reconcile Ephesus - Audit3 channel account handling

### ISSUE THREAD

**Issue by dobertRowneySr** [Joystream/joystream#3773]
##### Current logic
Reward account logic for claiming channel payouts

| Ephesus | Audit3 |
|----|----|
| reward account for channel is used | internal (sub)module account is used |

Reward account logic for NFT related proceeds (when NFT owner is a channel) and royalties
| Ephesus | Audit3 |
| ----------- | ----------- |
| reward account for channel is used (using the rules established in #3650 | internal (sub)module account is used |

TL;DR in Audit3 `channel::reward_account` is dropped in favour of sub module account, in `ephesus` old reward account logic is still used

#### Goal
Ephesus should have the same way of handling channel account as Audit3

Top priority

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202078489349362/1202312801565311) by [Unito](https://www.unito.io)
